src/libsystemd-network/network-internal.c

```diff
@@ -101,11 +102,51 @@ static bool net_condition_test_strv(char * const *patterns, const char *string)
         return has_positive_rule ? match : true;
 }
 
+static int net_condition_test_property(char * const *match_property, sd_device *device) {
+        char * const *p;
+
+        if (strv_isempty(match_property))
+                return true;
+
+        STRV_FOREACH(p, match_property) {
+                _cleanup_free_ char *buf = NULL;
+                const char *dev_val;
+                char *key, *val;
+                bool invert;
+
+                buf = strdup(*p);
+                if (!buf)
+                        return log_oom();
+
+                invert = *buf == '!';
+                key = buf + invert;
+                val = strchr(buf, '=');
+                if (!val)
+                        return -EINVAL;
+                *val++ = '\0';
+
+                if (invert) {
+                        if (device &&
+                            sd_device_get_property_value(device, key, &dev_val) >= 0 &&
+                            fnmatch(val, dev_val, 0) == 0)
+                                return false;
+                } else {
+                        if (!device ||
+                            sd_device_get_property_value(device, key, &dev_val) < 0 ||
+                            fnmatch(val, dev_val, 0) != 0)
+                                return false;
+                }
+        }
```

Comment:
This can be written as
```C
bool v = device && sd_device_get_property_value(device, key, &dev_val) >= 0 && fnmatch(val, dev_val, 0) == 0;
if (invert ? v : !v)
    return false;
```